crates/ra_hir/src/generics.rs

```diff
@@ -22,26 +22,62 @@ pub struct GenericParams {
     pub(crate) params: Vec<GenericParam>,
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Debug, Hash)]
+pub enum GenericDef {
```

Comment:
impl blocks can also have generics, which the old code didn't handle, but now I guess it could just be added here... :)

hm, though I guess impl blocks first need proper IDs instead of being an ID + arc...